server/conn.go

```diff
@@ -16,26 +16,22 @@ import (
 type Conn struct {
 	*packet.Conn
 
-	serverConf     *Server
-	capability     uint32
-	charset        uint8
-	authPluginName string
-	connectionID   uint32
-	status         uint16
-	warnings       uint16
-	salt           []byte // should be 8 + 12 for auth-plugin-data-part-1 and auth-plugin-data-part-2
-
 	credentialProvider  CredentialProvider
-	user                string
+	h                   Handler
+	stmts               map[uint32]*Stmt
+	serverConf          *Server
+	authPluginName      string
 	password            string
+	user                string
+	salt                []byte // should be 8 + 12 for auth-plugin-data-part-1 and auth-plugin-data-part-2
+	stmtID              uint32
```

Comment:
Could you keep `stmts` and `stmtID` near each other?

Reply:
While the config of binlog parsing shouldn't really put a hit on the memory usage when it isn't optimally aligned, especially `server.Conn` and `client.Conn` can make a real change in memory usage in a connection hungry application. I'd say this is the one struct you'd want to align as best as possible. I can add comments on what each field represents, perhaps that is the right middle ground?

Reply:
@atercattus what is your opinion about this?